PWGHF/TableProducer/candidateSelectorOmegac.cxx

```diff
@@ -0,0 +1,380 @@
+        continue;
+      }
+      if (candidate.dcaV0Dau() > dcaV0DauMax) {
+        continue;
+      }
+
+      // dca omegac daughters cut
+      if (candidate.dcaOmegacDau() > dcaOmegacDauMax) {
+        continue;
+      }
+
+      // cut on primary pion dcaXY
+      if ((candidate.dcaxyToPVPrimaryPi() < dcaxyPriPiMin) || (candidate.dcaxyToPVPrimaryPi() > dcaxyPriPiMax)) {
+        continue;
+      }
+
+      // pT selections
+      double ptPiFromCasc = RecoDecay::sqrtSumOfSquares(candidate.pxPiFromCascAtProd(), candidate.pyPiFromCascAtProd());
+      double ptPiFromOme = RecoDecay::sqrtSumOfSquares(candidate.pxPrimaryPiAtProd(), candidate.pyPrimaryPiAtProd());
+      if (std::abs(ptPiFromCasc) > ptMinPiFromCasc) {
+        continue;
+      }
+      if (std::abs(ptPiFromOme) > ptMinPiFromOme) {
+        continue;
+      }
+
+      //  TPC clusters selections
+      if (trackPiFromOmeg.tpcNClsFound() < nClustersTpcMin) {
+        continue;
+      }
+      if (trackPiFromLam.tpcNClsFound() < nClustersTpcMin) {
+        continue;
+      }
+      if (trackPrFromLam.tpcNClsFound() < nClustersTpcMin) {
+        continue;
+      }
+      if (trackPiFromCasc.tpcNClsFound() < nClustersTpcMin) {
+        continue;
+      }
+
+      if (trackPiFromOmeg.tpcNClsCrossedRows() < TpcCrossedRowsMin) {
+        continue;
+      }
+      if (trackPiFromLam.tpcNClsCrossedRows() < TpcCrossedRowsMin) {
+        continue;
+      }
+      if (trackPrFromLam.tpcNClsCrossedRows() < TpcCrossedRowsMin) {
+        continue;
+      }
+      if (trackPiFromCasc.tpcNClsCrossedRows() < TpcCrossedRowsMin) {
+        continue;
+      }
+
+      if (trackPiFromOmeg.tpcCrossedRowsOverFindableCls() < TpcCrossedRowsOverFindableClustersRatioMin) {
+        continue;
+      }
+
+      //  ITS clusters selection
+      if (trackPiFromOmeg.itsNCls() < nClustersItsMin) {
+        continue;
+      }
+      if (trackPiFromOmeg.itsNClsInnerBarrel() < nClustersItsInnBarrMin) {
+        continue;
+      }
+      if (trackPiFromCasc.itsNCls() < nClustersItsMin) {
+        continue;
+      }
+
+      // track-level PID selection
+      int pidProton = -999;
+      int pidPiFromLam = -999;
+      int pidPiFromCasc = -999;
+      int pidPiFromOme = -999;
+      if (PidTpcOnly) {
+        pidProton = selectorProton.getStatusTrackPIDTPC(trackPrFromLam);
+        pidPiFromLam = selectorPionFromV0.getStatusTrackPIDTPC(trackPiFromLam);
+        pidPiFromCasc = selectorPionFromCasc.getStatusTrackPIDTPC(trackPiFromCasc);
+        pidPiFromOme = selectorPionFromOme.getStatusTrackPIDTPC(trackPiFromOmeg);
+      } else if (PidTpcTofCombined) {
+        pidProton = selectorProton.getStatusTrackPIDAll(trackPrFromLam);
+        pidPiFromLam = selectorPionFromV0.getStatusTrackPIDAll(trackPiFromLam);
+        pidPiFromCasc = selectorPionFromCasc.getStatusTrackPIDAll(trackPiFromCasc);
+        pidPiFromOme = selectorPionFromOme.getStatusTrackPIDAll(trackPiFromOmeg);
+      }
```

Comment:
You should take care of the possibility that `PidTpcOnly` and `PidTpcTofCombined` might be both `true` or both `false`.